app/models/bookmark.rb

```diff
@@ -1,3 +1,10 @@
+class ControllerValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    controllers = ["dashboard"] + (ActiveRecord::Base.connection.tables.map(&:to_s) + Permission.resources.map {|x| x.tableize }).uniq
```

Comment:
Why isn't caching OK here?  The only reason I think I can see is that permissions may change, but only really in the test case where they're created through FG (or if they get cached before plugins are loaded).

In a general production environment, the plugins, tables, permissions ought to be static once the app is loaded, so caching I think ought to be alright.  If it's just the test environment then adding an invalidation method to Permission might be easiest.